cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -259,8 +260,43 @@ open3d::geometry::PointCloud PointCloud::ToLegacyPointCloud() const {
                 core::eigen_converter::TensorToEigenVector3dVector(GetPoints());
     }
     if (HasPointColors()) {
-        pcd_legacy.colors_ = core::eigen_converter::TensorToEigenVector3dVector(
-                GetPointColors());
+        bool dtype_is_supported_for_conversion = true;
+        double normalization_factor = 1.0;
+        core::Dtype point_color_dtype = GetPointColors().GetDtype();
+
+        if (point_color_dtype == core::Dtype::UInt8) {
+            normalization_factor =
+                    1.0 /
+                    static_cast<double>(std::numeric_limits<uint8_t>::max());
+        } else if (point_color_dtype == core::Dtype::UInt16) {
+            normalization_factor =
+                    1.0 /
+                    static_cast<double>(std::numeric_limits<uint16_t>::max());
+        } else if (point_color_dtype != core::Dtype::Float32 &&
+                   point_color_dtype != core::Dtype::Float64) {
+            utility::LogWarning(
+                    "Dtype {} of color attribute is not supported for "
+                    "conversion to LegacyPointCloud and will be skipped. "
+                    "Supported dtypes include UInt8, UIn16, Float32, and "
+                    "Float64",
+                    point_color_dtype.ToString());
+            dtype_is_supported_for_conversion = false;
+        }
+
+        if (dtype_is_supported_for_conversion) {
+            if (normalization_factor != 1.0) {
+                core::Tensor rescaled_colors =
+                        GetPointColors().To(core::Dtype::Float64, true) *
```

Comment:
`/*copy=*/ true` might not be required here, as if `GetPointColors` is not `Float64` it will return a copy, and anyways `/` op will return a copy. So, using `true` here will make a double copy.

Reply:
* `* op`

Reply:
@reyanshsolis That is a good point. Tensor::Mul creates a new tensor. I reverted to `To` without the copy.

Reply:
@stotko Although using the `std::numeric_limits` makes the intent clear, it's also very verbose and I lean toward agreeing with @prewettg that there are other ways of making the intent clear without the verbosity. Moreover, I think of `std::numeric_limits` as a tool for generic programming which is not happening here. Nonetheless, your suggestion is more consistent with the code in t/geometry and core so I'm going with `std::numeric_limits` for getting the max value.